platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/Light.java

```diff
@@ -0,0 +1,128 @@
+// This file is generated. Edit android/platform/scripts/generate-style-code.js, then run `make android-style-code`.
+
+package com.mapbox.mapboxsdk.style;
+
+import android.support.annotation.UiThread;
+import com.mapbox.mapboxsdk.style.layers.TransitionOptions;
+import com.mapbox.mapboxsdk.style.layers.PropertyValue;
+
+/**
+ * 
+ *
+ * @see <a href="https://www.mapbox.com/mapbox-gl-style-spec/#light>">The online documentation</a>
+ */
+@UiThread
+public class Light {
+
+  private long nativePtr;
+
+  /**
+   * Creates a Light.
+   *
+   * @param nativePtr pointer used by core
+   */
+  public Light(long nativePtr) {
+    this.nativePtr = nativePtr;
+  }
+
+  /**
+   * Set the Anchor property.
+   *
+   * @param String value of anchor
+   */
+  public void setAnchor(String anchor) {
+    nativeSetAnchor(new PropertyValue<>("anchor", anchor));
+  }
+
+  /**
+   * Set the Position property.
+   *
+   * @param Float[] value of position
+   */
+  public void setPosition(Float[] position) {
```

Comment:
Apparently the position isn't a 2D point, neither a 3D point:
>Position of the light source relative to lit (extruded) geometries, in [r radial coordinate, a azimuthal angle, p polar angle] where r indicates the distance from the center of the base of an object to its light, a indicates the position of the light relative to 0° (0° when `light.anchor` is set to `viewport` corresponds to the top of the viewport, or 0° when `light.anchor` is set to `map` corresponds to due north, and degrees proceed clockwise), and p indicates the height of the light (from 0°, directly above, to 180°, directly below).

I opted to expose 3 float params with names:
> float radialCoordinate, float azimuthalAngle, float polarAngle

This would make it more obvious to end developers what these float types mean.